metabase/metabase
# Fix - Clicking Saved Questions option in data selector when instance have no questions leads to Raw Data option instead

Closes #29760

### Description

There is partial duplication of this logic in `query_builder/components/DataSelector` & `containers/DataPicker`.
This PR tries to consolidate and reuse some of that logic from the latter in the former.

- Refactor `MetabaseSettings.get("enable-nested-queries")` in `DataSelector`
  - :warning: This affects how settings are mocked in tests, so if you try out the unit test that reproduces this on `master`, this test won't work as expected. The test marked with `(metabase#29760)` should correctly detect the reported issue.
- Refactor `getDataTypes` in `DataPicker/utils.ts`
  - move `DataTypeInfoItem` to `DataPicker/types.ts`
  - extract & move constants to `DataPicker/constants.ts`
  - improve test coverage
- Refactor `DataSelectorDataBucketPicker`
  - Move `DATA_BUCKET` to `metabase/containers/DataPicker`
  - Reuse `getDataTypes` from `metabase/containers/DataPicker`
  - Remove props spreading
  - Remove `Bucket` type in favor of `DataTypeInfoItem`
  - Fix typing of `onChangeDataBucket` prop
- Refactor `useDataPickerValue` to use named export

`query_builder/components/DataSelector` implementation is very complex and has lots of issues. I already found and reported 2 different ones while working on it:
- #32252
- #32225

### How to verify

0. Create a fresh Metabase instance.
1. Create a new model and save it
2. Click `+ New` -> `Question`

There should be no "Saved Questions" option visible in the popup.


### Demo

#### Before

https://github.com/metabase/metabase/assets/6830683/886fd8f7-1902-4b66-b88b-fea67c9a197b

#### After

https://github.com/metabase/metabase/assets/6830683/ebef0ad3-63b0-4510-ba78-de149963e1be
